src/katgpucbf/fgpu/engine.py

```diff
@@ -377,11 +377,30 @@ def capacity(self) -> int:  # noqa: D401
         # basically, the number of spectra.
         return self.spectra.shape[0] * self.spectra.shape[2]
 
+    @property
+    def next_timestamp(self) -> int:  # noqa: D401
+        """Timestamp of the next chunk afer this one."""
+        return self.timestamp + self.capacity * self.spectra_samples
+
     @property
     def pols(self) -> int:  # noqa: D401
         """Number of polarisations."""
         return self.spectra.shape[3]
 
+    @property
+    def all_present(self) -> bool:  # noqa: D410
+        """All batches for this chunk are complete.
+
+        :attr:`OutQueueItem.present` does not get "reset" in-between chunks. If
+        one chunk is complete and it is filled with the value `True`, and the
+        last few heaps / batches of the next chunk processed by this Item go
+        missing, you won't be able to tell directly from _out_item.present.
```

Comment:
This is talking about implementation details rather than semantics. I'd rather say something like (you'll probably need to reflow it):
```suggestion
        Only the first :attr:`n_spectra` elements of :attr:`OutQueueItem.present` have
        defined values, so it is not safe to use ``np.all(item.present)``.
```